buildSrc/src/main/groovy/org/opensearch/gradle/CheckCompatibilityTask.groovy

```diff
@@ -43,36 +43,44 @@ class CheckCompatibilityTask extends DefaultTask {
         repositoryUrls.parallelStream().forEach { repositoryUrl ->
             logger.lifecycle("Checking compatibility for: $repositoryUrl with ref: $ref")
             def tempDir = File.createTempDir()
+            def stdout = new ByteArrayOutputStream()
+            def errout = new ByteArrayOutputStream()
+            def skipped = false;
             try {
                 if (cloneAndCheckout(repositoryUrl, tempDir)) {
                     if (repositoryUrl.toString().endsWithAny('notifications', 'notifications.git')) {
                         tempDir = Paths.get(tempDir.getAbsolutePath(), 'notifications')
                     }
                     project.exec {
                         workingDir = tempDir
-                        def stdout = new ByteArrayOutputStream()
                         executable = (OperatingSystem.current().isWindows()) ? 'gradlew.bat' : './gradlew'
-                        args 'assemble'
+                        args ('assemble')
                         standardOutput stdout
+                        errorOutput errout
                     }
                     compatibleComponents.add(repositoryUrl)
                 } else {
-                    logger.lifecycle("Skipping compatibility check for $repositoryUrl")
+                    skipped = true
                 }
             } catch (ex) {
                 failedComponents.add(repositoryUrl)
                 logger.info("Gradle assemble failed for $repositoryUrl", ex)
             } finally {
+                if (skipped) {
+                    logger.lifecycle("Skipping compatibility check for $repositoryUrl")
+                } else {
+                    logger.lifecycle("Finished compatibility check for $repositoryUrl")
```

Comment:
I cloned this branch and tried running the check looks like line 72 is getting written between logs. Example:

```
/tmp/groovy-generated-tmpdir-6789044170191388467/build/generated/sources/delombok/java/main/org/opensearch/knn/common/KNNConstants.java:24: warning: no comment
    public static final String CFinished compatibility check for https://github.com/opensearch-project/security.git
Finished compatibility check for https://github.com/opensearch-project/cross-cluster-replication.git
```
Another example:
```
e: file:///tmp/groovy-generated-tmpdir-15826201936636853589/src/main/kotlin/org/opensearch/replication/util/Coroutines.kt:89:21 Type mismaFinished compatibility check for https://github.com/opensearch-project/job-scheduler.git
tch: inferred type is Resp! but ActionResponse! was expected
e: file:///tmp/groovy-generated-tmpdir-15826201936636853589/src/main/kotlin/org/opensearch/replication/util/Coroutines.kt:89:34 Type mismatch: inferred type is CoroutineActionListener<Resp> but ActionListener<TypeVariable(Response)!>! was expected
```
Any idea why?

Reply:
My solution is imperfect, each log system can be interrupted by others sine there is only one console to write too even across threads, but the likelihood goes down if they are novel failures, e.g. (completion after 1 second, 5 seconds and 20 seconds).

Reply:
Could you link to your workflow so I could take a look if there is anything you'd like to see changed?